extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java

```diff
@@ -593,25 +604,28 @@ public void handle(Void event) {
 
     static class QuartzTrigger implements Trigger {
 
-        final org.quartz.Trigger trigger;
+        final org.quartz.TriggerKey triggerKey;
+        final Function<TriggerKey, org.quartz.Trigger> triggerFunction;
         final ScheduledInvoker invoker;
         final Duration gracePeriod;
 
-        QuartzTrigger(org.quartz.Trigger trigger, ScheduledInvoker invoker, Duration gracePeriod) {
-            this.trigger = trigger;
+        QuartzTrigger(org.quartz.TriggerKey triggerKey, Function<TriggerKey, org.quartz.Trigger> triggerFunction,
+                ScheduledInvoker invoker, Duration gracePeriod) {
+            this.triggerKey = triggerKey;
+            this.triggerFunction = triggerFunction;
             this.invoker = invoker;
             this.gracePeriod = gracePeriod;
         }
 
         @Override
         public Instant getNextFireTime() {
-            Date nextFireTime = trigger.getNextFireTime();
+            Date nextFireTime = getTrigger().getNextFireTime();
```

Comment:
I wonder if we should swith to the original behavio before https://github.com/quarkusio/quarkus/commit/c4ac882fb3df5e849fcb06efeeed192c1bd8f39e, i.e. use the `JobExecutionContext` to obtain the current trigger instead of `triggerFunction`. And keep the `triggerKey` for the `getId()` method.

Reply:
I'll leave that you and @macchi1990 to decide

Reply:
I do not think it is it possible to use the `JobExecutionContext` here. 
Instances of this class are created way before `JobExecutionContext`, which is available during invocation. 
I could be missing something though.